### ISSUE TITLE
cuda-enabled jaxlib is coming to conda-forge!

### ISSUE THREAD

**Issue by ngam** [google/jax#10708]
We've built jaxlib 0.3.7 on conda-forge with cuda: https://github.com/conda-forge/jaxlib-feedstock/pull/97

I am opening this issue for viz but also for testing/confirmation all is good. Could someone from the team here have a look with us? Easy things to check, we get something like ~11000 actions/cycles of bazel compilation (compared to ~7500 for non-cuda) and the size of our artifact is similar to that of the official jaxlib ~200 mb. I tested it locally and it seems to put the arrays on the GPU by default. So it is working, but more confirmation is always good!

*Note* that we build for 10.2, 11.0, 11.1, and 11.2 on conda-forge by default, though we we can adjust this if requested, and do *note* the forward compatibility of 11.2, it is really "11.2+" meaning that it compiles for 11.2 and above (if someone installs that the 11.2 compiled version, it will in fact install cudatoolkit 11.6 with it and the latest cudnn)

People interested in checking this out, should head to the:
- PR: link above
- logs: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=506049
- artifacts: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=506049&view=artifacts&pathAsName=false&type=publishedArtifacts

You will see we still have a few problems (lost connections + timeouts mainly) due to the strict 6-hour time limit (it is tricky because we are very close to the limit, but hopefully we will manage to get all of them by rerunning the CI a few times)

Tagging @hawkinsp again who's been involved in issues related to the builds before + @jakevdp who's been very helpful and engaging.

---

Please:

- [x] Check for duplicate requests.
- [x] Describe your goal, and if possible provide a code snippet with a motivating example.

**Comment by ngam** [comment]
Thanks @hawkinsp, dropping 11.0 and lower. 

Seems like unvendoring nccl is as simple as setting an env variable, let's see if this reduces the build time significantly enough to fit in the 6-hour limit.... 
https://github.com/tensorflow/tensorflow/blob/5dcfc51118817f27fad5246812d83e5dccdc5f72/third_party/nccl/nccl_configure.bzl#L5

**Comment by ngam** [comment]
Closing as the PR was reviewed and merged, we opted to release only cuda 11.2+ as recommended here. 

One note: We have a tricky constraint to avoid polluting people's environments with large cuda binaries when unneeded, so one would need to override this constraint if on a machine with a GPU, but would still like to get the cuda-enabled jax (e.g. on a login node of an HPC):

```
CONDA_OVERRIDE_CUDA="11.2" conda create -n cudajax jaxlib==*=*cuda* jax -c conda-forge
```

Feel free to reach out to us in the future at conda-forge/jaxlib-feedstock if there are any issues. We hope this is a good addition to the community!

**Comment by hawkinsp** [comment]
BTW, we have some tentative plans to split up `jaxlib` into submodules:
* `jaxlib`: contains common infrastructure and the CPU platform
* `jaxlib-cuda`: contains the CUDA platform. Depends on `jaxlib`.
etc.

i.e., the CUDA wheel would be an extension to the CPU wheel, not an independent build. I think that may resolve the "tricky constraint" above.
